components/builder-depot/src/server.rs

```diff
@@ -639,10 +639,16 @@ fn download_latest_origin_secret_key(req: &mut Request) -> IronResult<Response>
         }
         Err(err) => return Ok(render_net_error(&err)),
     }
-    match route_message::<OriginSecretKeyGet, OriginSecretKey>(req, &request) {
-        Ok(ref key) => Ok(render_json(status::Ok, key)),
-        Err(err) => Ok(render_net_error(&err)),
-    }
+    let key = match route_message::<OriginSecretKeyGet, OriginSecretKey>(req, &request) {
+        Ok(key) => key,
+        Err(err) => {
+            error!("Can't retrieve secret key file: {}", err);
+            return Ok(render_net_error(&err));
+        }
+    };
+
+    let xfilename = format!("{}-{}.sig.key", key.get_name(), key.get_revision());
```

Comment:
Is this key formatting logic present anywhere else in the codebase? It'd be nice to reuse if possible.

Reply:
I grepped through and the places I found that were outputting keys were formatting things manually, as here.  It'd be nice to consolidate all that into one place, but I think that's out of scope for this PR.